packages/belos/src/BelosSolverFactory.hpp

```diff
@@ -74,7 +58,46 @@
 #include <stdexcept>
 #include <vector>
 
+// For convenience have a specialized form of registerSolverFactoryForLib()
+// called at construction of SolverFactoryParent which will pick up the proper
+// lib (belos, belosepetra, or belosepetra) based on the template parameters.
+// Then registerSolverFactoryForLib() will link to and register all the managers
+// for the proper lib. These includes were added for the specialization.
+
+// for belos lib
+#include "BelosMultiVec.hpp"
+#include "BelosOperator.hpp"
+
+// for belosepetra lib
+#ifdef HAVE_BELOS_EPETRA
+#include "Epetra_MultiVector.h"
+#include "Epetra_Operator.h"
+#endif
+
+// for belostpetra lib
+#ifdef HAVE_BELOS_TPETRA
```

Comment:
Hm.... I think this is OK only because Belos does not have a subpackage architecture.  Belos itself has an optional dependency on Tpetra; there is no "BelosTpetra subpackage."  Ditto for Epetra and Thyra.  (The problem with Thyra is that the Belos adapters actually live in a different package -- I think Stratimikos -- though that could be fixed.)

Reply:
Note I did not do any special effort on Thyra yet so perhaps there is further stuff I need to enable to see issues that need to be resolved.